protostar/migrator/migrator_contract_identifier_resolver.py

```diff
@@ -0,0 +1,32 @@
+from pathlib import Path
+
+from protostar.compiler import CompiledContractWriter, ProjectCompiler
+from protostar.migrator.migrator_datetime_state import MigratorDateTimeState
+
+
+class MigratorContractIdentifierResolver:
+    def __init__(
+        self,
+        project_compiler: ProjectCompiler,
+        migrator_datetime_state: MigratorDateTimeState,
+    ) -> None:
+        self._project_compiler = project_compiler
+        self._migrator_datetime_state = migrator_datetime_state
+
+    def resolve(self, contract_identifier: str) -> Path:
+        if "." in contract_identifier:
+            return Path(contract_identifier)
+        return self._compile_contract_by_contract_name(contract_identifier)
+
+    def _compile_contract_by_contract_name(self, contract_name: str) -> Path:
```

Comment:
Are we logging that the contract is being compiled at this moment?

Reply:
Nope, I'll add this

- [ ] log compiling

Reply:
Logging the compilation step requires extracting logging logic out of the gateway facade. I created a task for that: 
https://github.com/software-mansion/protostar/issues/854